packages/dds/tree/src/simple-tree/treeNodeValid.ts

```diff
@@ -157,8 +157,8 @@ export abstract class TreeNodeValid<TInput> extends TreeNode {
 		);
 
 		const result = schema.prepareInstance(this, node);
-		new TreeNodeKernel(result, schema);
-		setInnerNode(result, node);
+		new TreeNodeKernel(result, schema, node);
```

Comment:
I know you didn't introduce this, but a floating constructor call makes me want to ask if this is code smell.

Reply:
I don't really like it, but there isn't a clear more robust API I could think of (I considered several options when I originally did this), and this is literally the only place that class will ever be instanced, so it seems fine though odd.

Given pretty much all the other code in here is deceptively fragile and kinda horrifying (ex: preceding line does static method virtual dispatch on a base class blind casted from the this object's constructor parameter to produce whats usally a proxy which is returned from the constructor to impersonate one of the base classes.....) having one scary looking thing thats actually totally safe and robust next to all the actually scary things might not be too bad.

Reply:
I have add a comment to indicate this was intended.